components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubCorrelationLogUtils.java

```diff
@@ -0,0 +1,142 @@
+
+        if (isCorrelationLogsEnabled() && correlationLog.isInfoEnabled()) {
+            List<String> logPropertiesList = new ArrayList<>();
+            logPropertiesList.add(CORRELATION_LOG_REQUEST_START);
+            logPropertiesList.add(Long.toString(System.currentTimeMillis()));
+            logPropertiesList.add(request.getMethod());
+            logPropertiesList.add(request.getURI().getQuery());
+            logPropertiesList.add(request.getURI().getPath());
+
+            correlationLog.info(createFormattedLog(logPropertiesList));
+        }
+    }
+
+    /**
+     * Trigger correlation logs for http out response.
+     *
+     * @param request           Http out request.
+     * @param requestStartTime  Request start time.
+     * @param otherParams       Other response parameters that needs to be logged.
+     */
+    public static void triggerCorrelationLogForResponse(HttpEntityEnclosingRequestBase request, long requestStartTime,
+                                                        String... otherParams) {
+
+        if (isCorrelationLogsEnabled() && correlationLog.isInfoEnabled()) {
+            long currentTime = System.currentTimeMillis();
+            long timeTaken = currentTime - requestStartTime;
+
+            List<String> logPropertiesList = new ArrayList<>();
+            logPropertiesList.add(Long.toString(timeTaken));
+            logPropertiesList.add(CORRELATION_LOG_REQUEST_END);
+            logPropertiesList.add(Long.toString(requestStartTime));
+            logPropertiesList.add(request.getMethod());
+            logPropertiesList.add(request.getURI().getQuery());
+            logPropertiesList.add(request.getURI().getPath());
+            Collections.addAll(logPropertiesList, otherParams);
+
+            correlationLog.info(createFormattedLog(logPropertiesList));
+        }
+    }
+
+    /**
+     * Is correlation logs enabled in the system.
+     *
+     * @return Boolean indicating correlation logs enabled or not.
+     */
+    private static boolean isCorrelationLogsEnabled() {
+
+        if (isEnableCorrelationLogs == null) {
+            isEnableCorrelationLogs = Boolean.parseBoolean(System.getProperty(CORRELATION_LOG_SYSTEM_PROPERTY));
+        }
+        return isEnableCorrelationLogs;
+    }
+
+    /**
+     * Create the log line that should be printed.
+     *
+     * @param logPropertiesList List of log values that should be printed in the log.
+     * @return The log line.
+     */
+    private static String createFormattedLog(List<String> logPropertiesList) {
+
+        StringBuilder sb = new StringBuilder();
+        int count = 0;
+        for (String property: logPropertiesList) {
+            sb.append(property);
+            if (count < logPropertiesList.size() - 1) {
+                sb.append(CORRELATION_LOG_SEPARATOR);
+            }
+            count++;
+        }
+        return sb.toString();
```

Comment:
check this https://stackoverflow.com/a/22577565
```suggestion
        return String.join(CORRELATION_LOG_SEPARATOR, logPropertiesList);
```